src/internal/s3.ts

```diff
@@ -269,6 +269,41 @@ export class S3Client extends AWSClient {
         this._handle_error('DeleteObject', res)
     }
 
+    /**
+     * Copies an object from one bucket to another
+     *
+     * @param  {string} sourceBucket - The source bucket name containing the object.
+     * @param  {string} sourceKey - Key of the source object to copy.
+     * @param  {string} destinationBucket - The destination bucket name containing the object.
+     * @param  {string} destinationKey - Key of the destination object.
+     * @throws  {S3ServiceError}
+     * @throws  {InvalidSignatureError}
+     */
+    copyObject(sourceBucket: string, sourceKey: string, destinationBucket: string, destinationKey: string): void {
+        // Prepare request
+        const method = 'PUT'
+        const host = `${destinationBucket}.${this.host}`
+
+        const signedRequest = this.signature.sign(
+            {
+                method: method,
+                protocol: this.scheme,
+                hostname: host,
+                path: `/${destinationKey}`,
+                headers: {
+                    'x-amz-copy-source': `${sourceBucket}/${sourceKey}`,
+                },
+            },
+            {}
+        )
+
+        const res = http.request(method, signedRequest.url, signedRequest.body || null, {
```

Comment:
I'm interested to get some additional thoughts on this line. The other methods default the request body to an empty string instead of null. When using null, things work fine, but when defaulting to `''` I saw the following error when using `copyObject` in my test:

```
ERRO[0002] Uncaught (in promise) S3ServiceError: A header you provided implies functionality that is not implemented
```

Reply:
TL;DR: `body || null` should be the default from now on, so, you did good 👍🏻 

The `body || ''` is legacy, we just didn't get to fix it yet unfortunately. We're gonna prioritize that. In general, a reason we should prefer to use `null` instead of `''` is that it avoids getting in the way of the signature process (AWS verifies the signature partly based on the hash of the body, and explicitly setting the body was a way for us to make sure that if no body was present, we would calculate the same hash than AWS).